pkg/virt-controller/watch/vm.go

```diff
@@ -2732,12 +2733,17 @@ func (c *VMController) handleDynamicInterfaceRequests(vm *virtv1.VirtualMachine,
 			return err
 		}
 
+		updatedVMIfaces := vmispec.IndexInterfaceSpecByName(vm.Spec.Template.Spec.Domain.Devices.Interfaces)
 		vmiSpecCopy := vmi.Spec.DeepCopy()
 		for i := range vm.Status.InterfaceRequests {
 			request := &vm.Status.InterfaceRequests[i]
 			switch {
 			case request.AddInterfaceOptions != nil:
 				vmiSpecCopy = controller.ApplyNetworkInterfaceAddRequest(vmiSpecCopy, request.AddInterfaceOptions)
+
+				if iface := vmispec.LookupInterfaceByName(vmiSpecCopy.Domain.Devices.Interfaces, request.AddInterfaceOptions.Name); iface != nil {
```

Comment:
That change at this stage is a noop.
I guess the correct order should have been: First split the VM and VMI handling, one placed after the VM update. Then add the mac treatment (this patch).

But I will not insist on the order at this stage.